Mage.Sets/src/mage/cards/a/AminatousAugury.java

```diff
@@ -134,54 +166,151 @@ public AminatousAuguryCastFromExileEffect copy() {
+        this.text = "for each nonland cardtype, you may cast a card of that type from among the exiled cards without paying its mana cost";
+    }
+
+    AminatousAuguryCost(final AminatousAuguryCost cost) {
+        super(cost);
+    }
+
+    @Override
+    public boolean pay(Ability ability, Game game, UUID sourceId, UUID controllerId, boolean noMana, Cost costToPay) {
+
+        /*
+         Use map to match card to the Aminatou card used to exile it. Then use that Aminatou card Id,
+         to load castable card types
+        */
+        HashMap<UUID,UUID> setLocater = (HashMap<UUID, UUID>) game.getState().getValue("cardToAminatouIdMap");
+        String aminatouId = setLocater.get(sourceId).toString() + "_unusedCardTypes";
+        EnumSet<CardType> unusedCardTypes = (EnumSet<CardType>) game.getState().getValue(aminatouId);
+        if (unusedCardTypes == null){
+            throw new UnsupportedOperationException("Could not load unused card types for Aminatou's Augury");
+        }
+
+        Set<CardType> castCardTypes = ability.getSourceObject(game).getCardType();
+        if (castCardTypes != null) {
+
+            // Count matching available card types
+            CardType useType = null;
+            int numAvailTypes = 0;
+            for (CardType chkType : castCardTypes){
+                if (unusedCardTypes.contains(chkType)){
+                    numAvailTypes++;
+                    useType = chkType;
+                }
+            }
+
+            if (numAvailTypes == 0){
+                // If no available types, then do not cast.
+                paid = false;
+            } else if (numAvailTypes == 1){
+                // If card has multiple types, and only one is unused,
+                // then cast using that type OR if card has only one type
+                // and it is available, then cast.
+                unusedCardTypes.remove(useType);
+                paid = true;
+            } else if (numAvailTypes >= 2){
+                // If card has multiple types, and more than one has not yet been used,
+                // then show dialog to select which type to use for cast.
+                AminatousAuguryCardTypeChoice choices = new AminatousAuguryCardTypeChoice(castCardTypes);
+                Player controller = game.getPlayer(controllerId);
+                if (controller.choose(Outcome.Neutral, choices, game)) {
+                    CardType choiceType = CardType.fromString(choices.getChoice());
+                    unusedCardTypes.remove(choiceType);
+                    paid = true;
+                }
+            }
+        } else {
+            throw new UnsupportedOperationException("Card type returned null.");
+        }
+
+
+        game.getState().setValue( aminatouId, unusedCardTypes);
+
+        return paid;
+    }
+
+
+    @Override
+    public boolean canPay(Ability ability, UUID sourceId, UUID controllerId, Game game) {
+        return targets.canChoose(controllerId, game);
```

Comment:
Wrong `canPay` method -- you must find current card, card's types and compare with used types list. Is it can pay or not. And return result here. Look above comment with `findUnusedList` example (you use same method to find unused types in multiple places).